app/controllers/api/v1/locations_controller.rb

```diff
@@ -34,6 +34,18 @@ def update
     end
   end
 
+  # PUT /locations/pickup_day
+  # updates pickup day for all locations at once
+  def update_pickup_day
+    if valid_pickup_day?(params[:pickup_day])
```

Comment:
If you want to keep validation logic within the data model, you could have a `validates :pickup_date, inclusion: Date::DAYNAMES` and you could get the error message for _free_.